Python/thread_pthread.h

```diff
@@ -29,7 +29,9 @@
    be conditional on _POSIX_THREAD_ATTR_STACKSIZE being defined. */
 #ifdef _POSIX_THREAD_ATTR_STACKSIZE
 #ifndef THREAD_STACK_SIZE
-#define THREAD_STACK_SIZE       0       /* use default stack size */
+/* -1 means THREAD_STACK_SIZE hasn't been given a proper value yet.
+   -DTHREAD_STACK_SIZE=0 in CFLAGS will force the platform's default instead. */
```

Comment:
to avoid disruption for anyone who may already be supplying CFLAGS containing -DTHREAD_STACK_SIZE=0 in their builds already keep the existing 0 behavior: 0 should be the platform specific values below.

If we want a new value in some circumstances, that should either be enabled via an appropriate test in `configure.ac` and thus via pyconfig.h. Or yet another platform specific compile time `#if` check should be used here to detect musl libc.

Reply:
> Or yet another platform specific compile time #if check should be used here to detect musl libc.

To be clear, the purpose here is to make the code universal. Right now, the only platform where the default thread size actually works is glibc. Instead of adding a case for each and every platform other than glibc, IMO it makes much more sense to have a sane default and override for one or two platforms as necessary, instead.

Testing for musl is unsupported, they don't provide any identifying macros, by design.